clinica/iotools/converters/ixi_to_bids/ixi_to_bids_utils.py

```diff
@@ -0,0 +1,449 @@
+    return pd.DataFrame({"SEX": ["male", "female"]}, index=[1, 2])["SEX"]
+
+
+class Clinical_Data_Mapping(str, Enum):
+    ETHNIC = "Ethnicity"
+    MARITAL = "Marital Status"
+    OCCUPATION = "Occupation"
+    QUALIFICATION = "Qualification"
+
+
+def _get_mapping(clinical_data_path: Path, map: Clinical_Data_Mapping) -> pd.DataFrame:
+    try:
+        df = pd.read_excel(
+            clinical_data_path / "IXI.xls", sheet_name=map.value
+        ).set_index("ID")[map.name]
+    except FileNotFoundError:
+        log_and_raise(
+            f"Clinical data stored in the folder {clinical_data_path} is expected to be an excel file named 'IXI.xls'. "
+            f"In case the file downloaded from the IXI website changed format, please do not hesitate to report to us !",
+            FileNotFoundError,
+        )
+    except (ValueError, KeyError):
+        log_and_raise(
+            f"{map.value} mapping is expected to be contained in a sheet called {map.value} coming from the clinical data excel. "
+            f"Possibilities are supposed to be described in a {map.name} column associated to keys from the 'ID' column. "
+            f"In case the file downloaded from the IXI website changed format, please do not hesitate to report to us !",
+            ValueError,
+        )
+    else:
+        return df
+
+
+def _padding_source_id(source_id: Union[str, int]) -> str:
+    if len(str(source_id)) > 3:
+        log_and_raise(
+            f"The source id {source_id} has more than 3 digits while IXI"
+            f"source ids are expected to be between 1 and 3 digits.",
+            ValueError,
+        )
+    return f"IXI{'0'* (3 - len(str(source_id))) + str(source_id)}"
+
+
+def read_clinical_data(clinical_data_path: Path) -> pd.DataFrame:
+    """
+    Reads and formats IXI clinical data.
+
+    Parameters
+    ----------
+    clinical_data_path : Path to the directory where the clinical data .xls is stored.
+
+    Returns
+    -------
+    A dataframe containing the clinical data, with some modifications: padded study id and added session id.
+
+    """
+    try:
+        clinical_data = pd.read_excel(clinical_data_path / "IXI.xls")
+    except FileNotFoundError:
+        log_and_raise(
+            f"Clinical data stored in the folder {clinical_data_path} is expected to be an excel file named 'IXI.xls'. "
+            f"In case the file downloaded from the IXI website changed format, please do not hesitate to report to us !",
+            FileNotFoundError,
+        )
+    else:
+        clinical_data.drop("DATE_AVAILABLE", axis=1, inplace=True)
+        clinical_data["SEX_ID (1=m, 2=f)"] = clinical_data["SEX_ID (1=m, 2=f)"].map(
+            _get_sex_mapping()
+        )
+
+        for column in ("ETHNIC_ID", "MARITAL_ID", "OCCUPATION_ID", "QUALIFICATION_ID"):
+            clinical_data[column] = clinical_data[column].map(
+                _get_mapping(
+                    clinical_data_path, Clinical_Data_Mapping[column.replace("_ID", "")]
```

Comment:
You can use your enumeration here:

```python
for mapping in ClinicalDataMapping:
    clinical_data[f"{mapping.name}_ID"] = clinical_data[f"{mapping.name}_ID"].map(
        _get_mapping(clinical_data_path, mapping)
    )
```

Reply:
I think this is still relevant

Reply:
I think this is the last remaining point. It's not a major one though. Let me know what you want to do about it.

Reply:
I just discovered the button in git to show unresolved conversations >< will apply this right away !